include/platform/mir/renderer/sw/pixel_source.h

```diff
@@ -30,6 +35,74 @@ namespace renderer
 namespace software
 {
 
+class BufferDescriptor
```

Comment:
Candidates for a better name for `BufferDescriptor` are most welcome!

Reply:
Well, this appears to simply be some properties a has Buffer. So "Buffer"? Or "GenericBuffer"?